### PR TITLE
Some additional fixes for Xcode 15

### PR DESCRIPTION
**Pull request by StephaneMagne**
I also updated the RemoteStore logic to be closer to the previous Combine logic for consistency.

### CODE REVIEW

File: LucidTests/Core/CoreManagerTests.swift [hunk 1]
```diff
@@ -8728,7 +8728,7 @@ final class CoreManagerTests: XCTestCase {
         coreDataStore.set(entities, in: WriteContext(dataTarget: .local, accessValidator: nil)) { _ in
             setExpectation.fulfill()
         }
-        wait(for: [setExpectation], timeout: 1)
+        await fulfillment(of: [setExpectation], timeout: 1)
```
Comment (Eze-Scribd) [inline LucidTests/Core/CoreManagerTests.swift:8731]: Nice, it was time for something like this :D

File: LucidTests/Stores/RecoverableStoreTests.swift [hunk 2]
```diff
@@ -45,22 +45,24 @@ final class RecoverableStoreTests: StoreTests {
         ).storing
     }
 
-    override func tearDown() async throws {
-        let success = await StubCoreDataManagerFactory.shared.clearDatabase()
-        if success == false {
-            XCTFail("Did not clear database successfully.")
-        }
+    override func asyncTearDown(_ completion: @escaping () -> Void) {
```
Comment (Eze-Scribd) [inline LucidTests/Stores/RecoverableStoreTests.swift:48]: Why was this change required? I'm not fully understanding the difference but the previous version using the `override func tearDown() async throws` was working as expected. Any particular case why we need the completion one?

Comment (StephaneMagne) [reply LucidTests/Stores/RecoverableStoreTests.swift:48]: This might have just been a collision between our changes. I had written my code using that function. I'm not entirely sure it's necessary TBH, but I was too deep into the code to step back and evaluate.

File: Lucid/Stores/CoreDataStore.swift [hunk 3]
```diff
@@ -146,258 +77,368 @@ public final class CoreDataManager: NSObject {
...
+            description.shouldAddStoreAsynchronously = true
+            description.shouldInferMappingModelAutomatically = true
+            description.shouldMigrateStoreAutomatically = true
+            description.type = storeType.descriptionType
+            persistentContainer.persistentStoreDescriptions = [description]
 
-                if let error = error {
-                    Logger.log(.error, "\(CoreDataManager.self): Error while loading persistent stores: \(error).")
+            return try await withCheckedThrowingContinuation { continuation in
+                persistentContainer.loadPersistentStores { description, error in
```
Comment (Eze-Scribd) [inline Lucid/Stores/CoreDataStore.swift:370]: So sad Apple didn't add async conformances to these functions

**Review by Eze-Scribd — COMMENTED** [review]
I left a couple of questions. Nice work with the CoreDataManager 👍

**Review by Eze-Scribd — APPROVED** [review]
